parts/k8s/cloud-init/artifacts/cse_main.sh

```diff
@@ -180,6 +182,7 @@ ps auxfww > /opt/azure/provision-ps.log &
 if $FULL_INSTALL_REQUIRED; then
   applyCIS || exit $ERR_CIS_HARDENING_ERROR
 else
+  KUBERNETES_VERSION="${KUBERNETES_VERSION_NO_SUFFIX}"
```

Comment:
why is this needed? `${KUBERNETES_VERSION_NO_SUFFIX}` will be unbound if the environment isn't Azure Stack

Reply:
Needed because my understanding of your previous comments was to adjust the consumer code in order to keep the functions unchaged.

The unbound situation is an issue.

Reply:
A simple fix here would be to simply do this re-assignment only for Azure Stack. 

The only concern I have with this approach is that if new functions that rely on `KUBERNETES_VERSION` are introduced in CSE it might break Azure Stack to have the suffix still around. But I guess we can deal with that problem once it comes. Do you have any other ideas on how to make this more elegant?

Reply:
Given how the scripts are organized, I agree that overriding `KUBERNETES_VERSION` is probably not recommended. I pushed it because you suggested the change and I do not want to break your patterns/design.

However, things get more complicated because you also seem to dislike the idea of changing the functions. I think we won't be able to have it both ways.

> Do you have any other ideas on how to make this more elegant?

Another option is to pass parameters to the functions instead of relying on global variables like `KUBERNETES_VERSION`. That would be a bigger change though.

I can go either way, just let me know.